### DISCUSSION TITLE
Why might image quality be decreased in prod?

### DISCUSSION THREAD

**Discussion by eolesinski** [alshedivat/al-folio#2255] [Q&A] [answered]
I just deployed my al-folio GitHub pages theme the other day, but had a quick question. I initially developed locally and tested with a local server (the legacy way on the `Install.md` guide). When I deployed locally the images seemed very crisp. When I deployed the site to GitHub pages to go live the image quality seemed to be reduced. I have attached some screenshots from the projects section below.

Image from local deployment:

<img width="467" alt="local" src="https://github.com/alshedivat/al-folio/assets/47125809/1844ea68-3e7d-4eed-9bb1-a10d07200682">

Image from prod:

<img width="467" alt="prod" src="https://github.com/alshedivat/al-folio/assets/47125809/91f59e12-aaa1-4d01-a879-13069ce8352f">

Is there a specific setting that is doing this or is this just the way that this theme handles this? I am trying to figure out why this might be happening.

**Comment by george-gca** [comment, marked as answer]
If you used [figure.liquid](https://github.com/alshedivat/al-folio/blob/master/_includes/figure.liquid) to display the image, it internally uses [imagemagick](https://imagemagick.org/index.php) to create different resolutions of the same image. This is done to avoid serving a high resolution version of the image when it is not needed, so the page loads faster. For more information check [this guide about responsive images](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images).

You can disable this behavior in [_config.yml](https://github.com/alshedivat/al-folio/blob/ed2ac9655822ae7692d5537ea2f932a9d4004239/_config.yml#L345), although this is not usually recommended.

**Reply by eolesinski** [reply]
Interesting - thanks for the clarification + links. Since I am still somewhat of a novice when it comes to web dev - is it best practice to have lower resolution images served, even in this case, for performance reasons? I would have thought that it would have served a slightly higher resolution (since I am using my normal computer's web browser) and then only served the lower resolution version while viewing on mobile.

**Edit:** I just saw your edit that disabling this is usually not recommended.

**Reply by george-gca** [reply]
To be honest I am not a prolific web dev also, but I believe this depends on the use case. For example, in your screenshot the image is being used as a thumbnail for the project. In this place it is probably being displayed at a lower size (for example, your original might be 1600x1200, but the size being displayed in that place is 400x200. In this case it doesn't make sense to serve a high res version of the image if it has do be downscaled to 400x200 to be displayed there.
